### PR TITLE
auto VIRT-298910

### PR DESCRIPTION
**Pull request by yanqzhan**
Automate VIRT-298910 - [Static route] Start virtual networks with static route defined.

### CODE REVIEW

**Comment by yanqzhan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.virtual_network.network.static_route.start_test: PASS (6.16 s)

File: libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+- virtual_network.network.static_route:
+    type = static_route
+    start_vm = "no"
+    variants:
+        - start_test:
+            net_name = "network_def"
+            net_attrs = {'name': net_name, 'forward': {'mode': 'nat'}, 'ips': [{'dhcp_ranges': {'attrs': {'start': '192.168.120.2', 'end': '192.168.120.254'}}, 'address': '192.168.120.1', 'netmask': '255.255.255.0'}, {'family': 'ipv6', 'address': '2001:db8:ca2:2::1', 'prefix': '64'}], 'routes': [{'address':'192.168.222.0','prefix':'24','gateway':'192.168.120.2'}, {'family':'ipv6','address':'2001:db8:ca2:3::','prefix':'64','gateway':'2001:db8:ca2:2::2'},{'family':'ipv6','address':'2001:db9:4:1::','prefix':'64','gateway':'2001:db8:ca2:2::3','metric':'2'}]}
+            ip_associated_route = '192.168.120.0/24'
+            ip_defined_route = '192.168.222.0/24'
+            ip6_associated_route = '2001:db8:ca2:2::/64'
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg:10]: ip6_associated_route-->ipv6_associated_route

File: libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- virtual_network.network.static_route:
+    type = static_route
+    start_vm = "no"
+    variants:
+        - start_test:
+            net_name = "network_def"
+            net_attrs = {'name': net_name, 'forward': {'mode': 'nat'}, 'ips': [{'dhcp_ranges': {'attrs': {'start': '192.168.120.2', 'end': '192.168.120.254'}}, 'address': '192.168.120.1', 'netmask': '255.255.255.0'}, {'family': 'ipv6', 'address': '2001:db8:ca2:2::1', 'prefix': '64'}], 'routes': [{'address':'192.168.222.0','prefix':'24','gateway':'192.168.120.2'}, {'family':'ipv6','address':'2001:db8:ca2:3::','prefix':'64','gateway':'2001:db8:ca2:2::2'},{'family':'ipv6','address':'2001:db9:4:1::','prefix':'64','gateway':'2001:db8:ca2:2::3','metric':'2'}]}
+            ip_associated_route = '192.168.120.0/24'
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg:8]: ip_associated_route -->ipv4_associated_route

File: libvirt/tests/src/virtual_network/network/elements_and_attributes/static_route.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ip_associated_route = params.get("ip_associated_route")
+    ip_defined_route = params.get("ip_defined_route")
+    ip6_associated_route = params.get("ip6_associated_route")
+    ip6_defined_routes = eval(params.get("ip6_defined_routes"))
+    try:
+        libvirt_network.create_or_del_network(net_attrs)
+        virsh.net_dumpxml(net_name, ignore_status=True)
+        net_info = virsh.net_info(net_name).stdout.strip()
+        bridge = re.search(r'Bridge:\s+(\S+)', net_info).group(1)
+        ip_route_cmd = "ip route | grep %s" % bridge
```
Comment (chunfuwen) [inline libvirt/tests/src/virtual_network/network/elements_and_attributes/static_route.py:25]: line 25 to 37 can be wrapped into one function, and function parameter may include have ipaddress_type(ipv4/ipv6), and other commands such as ip6_route_cmd  can be conmposble as per ipaddress_type

File: libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg (hunk 1 above)
Comment (yanqzhan) [reply libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg]: Done.

File: libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg (hunk 2 above)
Comment (yanqzhan) [reply libvirt/tests/cfg/virtual_network/network/elements_and_attributes/static_route.cfg]: Done.

File: libvirt/tests/src/virtual_network/network/elements_and_attributes/static_route.py (hunk 3 above)
Comment (yanqzhan) [reply libvirt/tests/src/virtual_network/network/elements_and_attributes/static_route.py]: Thanks. Done.

**Comment by yanqzhan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.virtual_network.network.static_route.start_test: PASS (6.15 s)

@chunfuwen Could you help review again please? Thank you!

**Review by chunfuwen — APPROVED** [review]
lgtm

**Comment by yanqzhan** [comment]
Modified per haijiao's suggestion. 
Rerun also pass:  (1/1) type_specific.io-github-autotest-libvirt.virtual_network.network.static_route.start_test: PASS (6.19 s)

**Review by chloerh — APPROVED** [review]

**Pull request merged**